src/core/lib/zlib/zutil.cc

```diff
@@ -3,12 +3,14 @@
  * For conditions of distribution and use, see copyright notice in zlib.h
  */
 #include "lib/zlib/zutil.h"
-namespace zlib {
 
-#pragma clang diagnostic push
-#pragma clang diagnostic ignored "-Wcast-qual"
-#pragma clang diagnostic ignored "-Wold-style-cast"
+namespace zlib {
 
+#if defined(__clang__)
```

Comment:
I think we should define a triple of macros `DT_COMPILER_CLANG`, `DT_COMPILER_GCC` and `DT_COMPILER_MSVC` in macros.h, similar to all other macros there. That way we'll have a more uniform and easy to remember spelling.

Reply:
Yeah, could be useful. I also found that it is very easy to miss an include like `utils/macros.h` and then to have an incorrect `#if DT_SOMETHING` behavior without any getting any warnings. Not sure what could be a good way around it.

Reply:
For me it produces a warning:
```
warning: 'DT_SOMETHING' is not defined, evaluates to 0 [-Wundef]
  #if DT_SOMETHING
```

It's actually `#ifdef S0METHING` that never produces a warning, even if you make a typo in the spelling of the macro. In this respect our current approach is safer.

Reply:
For some reason on Windows I never got the warning you post :/

Reply:
https://stackoverflow.com/questions/1643820/why-no-warning-with-if-x-when-x-undefined